pkg/network/encoding/encoding.go

```diff
@@ -52,17 +53,22 @@ func modelConnections(conns *network.Connections) *model.Connections {
 	domainSet := make(map[string]int)
 	routeIndex := make(map[string]RouteIdx)
 	httpIndex := FormatHTTPStats(conns.HTTP)
+	httpMatches := make(map[http.Key]struct{}, len(httpIndex))
 
 	for i, conn := range conns.Conns {
 		httpKey := httpKeyFromConn(conn)
-		agentConns[i] = FormatConnection(conn, domainSet, routeIndex, httpIndex[httpKey])
-		delete(httpIndex, httpKey)
+		httpAggregations := httpIndex[httpKey]
+		if httpAggregations != nil {
+			httpMatches[httpKey] = struct{}{}
+		}
+
+		agentConns[i] = FormatConnection(conn, domainSet, routeIndex, httpAggregations)
```

Comment:
Is the intent to have `httpAggregations` always be non-`nil`?

Reply:
httpAggregations will be `nil` only for connections that don't have HTTP traffic